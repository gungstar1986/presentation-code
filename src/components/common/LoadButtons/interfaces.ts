export type ButtonProps = {
  loadFn: () => void;
  title: string;
  isRender: boolean;
};
