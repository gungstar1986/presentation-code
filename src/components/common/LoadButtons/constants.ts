export const PREV_BUTTON = 'Load Prev';
export const NEXT_BUTTON = 'Load Next';
