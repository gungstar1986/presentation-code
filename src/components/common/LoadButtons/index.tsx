import React from 'react';

import { ButtonProps } from 'components/common/LoadButtons/interfaces';

export const LoadButtons: React.FC<ButtonProps> = ({ title, loadFn, isRender }) => {
  if (!isRender) return null;

  return (
    <div className="button-container">
      <button className="btn-default" onClick={loadFn}>
        {title}
      </button>
    </div>
  );
};
