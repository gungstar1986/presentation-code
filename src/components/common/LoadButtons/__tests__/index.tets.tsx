import React from 'react';

import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { LoadButtons } from 'components/common/LoadButtons/index';

describe('Button component', function () {
  it('should return Button component', function () {
    const loadFn = jest.fn();
    const title = 'title';
    const isRender = true;

    const wrapper = render(
      <BrowserRouter>
        <LoadButtons loadFn={loadFn} title={title} isRender={isRender} />
      </BrowserRouter>
    );

    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
