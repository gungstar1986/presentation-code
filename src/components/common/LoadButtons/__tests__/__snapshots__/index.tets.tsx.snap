// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should return Button component 1`] = `
<body>
  <div>
    <div
      class="button-container"
    >
      <button
        class="btn-default"
      >
        title
      </button>
    </div>
  </div>
</body>
`;
